results/cmd/api/main.go

```diff
@@ -170,16 +180,71 @@ func (s *server) DeleteTaskRunResult(ctx context.Context, req *pb.DeleteTaskRunR
 	return nil, nil
 }
 
+// ListTaskRunsResult receives a ListTaskRunRequest from users and return to users a list of TaskRuns according to the query
+func (s *server) ListTaskRunsResult(ctx context.Context, req *pb.ListTaskRunsRequest) (*pb.ListTaskRunsResponse, error) {
+	// Set up environment for cel and check if filter is empty string
+	ast, issues := s.env.Compile(req.GetFilter())
+	if issues != nil && issues.Err() != nil && req.GetFilter() != "" {
+		log.Printf("type-check error: %s", issues.Err())
+		return nil, status.Errorf(codes.InvalidArgument, "No TaskRuns found for the query string due to wrong type in the query, please try to enter a query with correct type again: %v", issues.Err())
+	}
+	// get all taskruns from database
+	rows, err := s.db.Query("SELECT taskrunlog FROM taskrun")
+	if err != nil {
+		log.Printf("failed to query on database: %v", err)
+		return nil, status.Errorf(codes.InvalidArgument, "failed to parse the query string due to internal server error, please try to enter the query again: %v", err)
```

Comment:
InvalidArgument usually indicates things went wrong because of the user input, but here we aren't using the query string / any other user input.

We should change the status code and the error message, since an error here would indicate an issue with the server. Because of this, perhaps internal makes sense?

Reply:
done